File: nats_client.py
from logging import Logger
import nats
import config


class NatsClient:
    __instance: nats.NATS | None = None

    @classmethod
    async def get_instance(cls, logger: Logger) -> nats.NATS:
        if cls.__instance is None:
            cls.__instance = await nats.connect(
                servers=[f"nats://{config.NATS_HOST}:{config.NATS_PORT}"],
                name="fastppi-webhook-nats-client",
                max_reconnect_attempts=-1,
                reconnect_time_wait=2,
            )
            logger.debug("Connected to NATS server.")
        return cls.__instance

File: nats_client.go
package main

import (
	"fmt"
	"log"
	"sync"

	"github.com/nats-io/nats.go"
)

var (
	nc   *nats.Conn
	once sync.Once
)

func getNats(logger *log.Logger) (*nats.Conn, error) {
	var err error
	once.Do(func() {
		url := fmt.Sprintf("nats://%s:%s", NatsHost, NatsPort)
		nc, err = nats.Connect(url, nats.Name("go-webhook-nats-client"))
		if err == nil {
			logger.Println("Connected to NATS server")
		}
	})
	return nc, err
}
